src/workerd/io/actor-cache.c++

```diff
@@ -1960,17 +1983,15 @@ ActorCache::DeleteAllResults ActorCache::deleteAll(WriteOptions options) {
     auto& map = currentValues.get(lock);
 
     kj::Vector<kj::Own<Entry>> deletedDirty;
+    for (auto& entry: dirtyList) {
+      // Preserve the dirty list so we can run it before our delete all. Note that it's not a given
```

Comment:
> Note that it's not a given that all dirty list entries are in the map.

So is this a bug fix given that we previously only iterated over the map? If so, what was the bug and why didn't we add a test reproducing it? Or was the code fine before and this "note" only applies to the code *after* the changes in this commit?

Reply:
> I would expect at least a clear example where a dirty entry isn't in `currentValues`

Going by the comments in the header file, that shouldn't be possible:

https://github.com/cloudflare/workerd/blob/2bf2d26a9ba5dd78868a25edb96f50b73c2740f9/src/workerd/io/actor-cache.h#L345-L346

Reply:
But your explanation makes sense, it seems that we're ok here and just need to clarify the comment.

Reply:
https://github.com/cloudflare/workerd/compare/ccaf59b65229b2520ad8e6f3913856471947d4ab..9bb1e4d593bf4e07382f72bdf28b2343716a8ab9